q2_demux/plugin_setup.py

```diff
@@ -228,4 +230,39 @@
                  'sequences after subsampling.')
 )
 
+T = TypeMatch([SequencesWithQuality, PairedEndSequencesWithQuality,
+               JoinedSequencesWithQuality])
+plugin.methods.register_function(
+    function=q2_demux.filter_samples,
+    inputs={'samples': SampleData[T]},
+    parameters={'metadata': Metadata,
+                'where': Str,
+                'exclude_ids': Bool},
+    outputs=[
+        ('filtered_samples', SampleData[T])
+    ],
+    input_descriptions={
+        'samples': 'The demultiplexed data from which samples should be '
+        'filtered.'
+    },
+    parameter_descriptions={
+        'metadata': 'Sample metadata used with `where` parameter when '
+                    'selecting samples to retain, or with `exclude_ids` '
+                    'when selecting samples to discard.',
```

Comment:
This implies `where` must be used. I would indicate what it does without where first, and then mention where can be added as well.